wbooks/src/redux/book/reducer.ts

```diff
@@ -0,0 +1,39 @@
+import { Book } from '@interfaces/Book';
+
+import { actions } from './actions';
+
+const initialState = {
+  books: [],
+  booksLoading: false,
+  bookError: null
+};
+
+interface Action {
+  type: string;
+  payload: Book[];
+}
+
+export const bookReducer = (state = initialState, action: Action) => {
+  switch (action.type) {
+    case actions.GET_BOOKS:
+      return {
+        ...state,
+        booksLoading: true
+      };
+
+    case actions.GET_BOOKS_SUCCESS:
+      return {
+        ...state,
+        books: action.payload
```

Comment:
You forgot to set the loading in false and error in null

Reply:
Is it really necessary? I mean, it already pulls it from the initial state

Reply:
and with every run, the state gets initially assigned to initialState